app/src/main/java/com/pvp/app/ui/screen/calendar/MonthlyCalendarScreen.kt

```diff
@@ -0,0 +1,240 @@
+        onClickListener = { date ->
+            showDialog = true
+            selectedDate = date.date
+            dateTasks = date.tasks
+        }
+    )
+
+    if (showDialog) {
+        DayDialog(
+            date = selectedDate,
+            tasks = dateTasks,
+            onDismissRequest = { showDialog = false }
+        )
+    }
+}
+
+@Composable
+fun DayDialog(
+    date: LocalDate,
+    tasks: List<Task> = listOf(),
+    onDismissRequest: () -> Unit
+) {
+    Dialog(
+        onDismissRequest = { onDismissRequest() },
+        properties = DialogProperties(usePlatformDefaultWidth = false)
+    ) {
+        Day(name = date.toString(), tasks = tasks)
+    }
+}
+
+@Composable
+fun MonthlyCalendar(
+    days: Array<String>,
+    month: YearMonth,
+    dates: List<CalendarUiState.DateEntry>,
+    onClickListener: (CalendarUiState.DateEntry) -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp)
+    ) {
+        Row {
+            repeat(days.size) {
+                val item = days[it]
+                DayItem(item, modifier = Modifier.weight(1f))
+            }
+        }
+
+        Header(
+            month = month,
+        )
+
+        Content(
+            dates = dates,
+            onClickListener = onClickListener
+        )
+    }
+}
+
+@Composable
+fun Header(
+    month: YearMonth,
+    model: MonthlyCalendarViewModel = hiltViewModel()
+) {
+    Row {
+        IconButton(onClick = {
+            model.changeMonth(month.minusMonths(1))
+        }) {
+            Icon(
+                imageVector = Icons.AutoMirrored.Filled.KeyboardArrowLeft,
+                contentDescription = "Back"
+            )
+        }
+
+        Text(
+            text = month.getDisplayName(),
+            textAlign = TextAlign.Center,
+            style = MaterialTheme.typography.bodyLarge,
+            modifier = Modifier
+                .weight(1f)
+                .align(Alignment.CenterVertically)
+        )
+
+        IconButton(onClick = {
+            model.changeMonth(month.plusMonths(1))
+        }) {
+            Icon(
+                imageVector = Icons.AutoMirrored.Filled.KeyboardArrowRight,
+                contentDescription = "Forward"
+            )
+        }
+    }
+}
+
+@Composable
+fun DayItem(day: String, modifier: Modifier = Modifier) {
+    Box(modifier = modifier) {
+        Text(
+            text = day,
+            style = MaterialTheme.typography.bodyMedium,
+            color = MaterialTheme.colorScheme.primary,
+            modifier = Modifier
+                .align(Alignment.Center)
+                .padding(10.dp)
+        )
+    }
+}
+
+@Composable
+fun Content(
+    dates: List<CalendarUiState.DateEntry>,
+    onClickListener: (CalendarUiState.DateEntry) -> Unit,
+) {
+    Column {
+        var index = 0
+
+        repeat(6) {
+            if (index >= dates.size) {
+                Log.e("CONTENT DATE FILTER", dates.size.toString())
```

Comment:
Single empty line required before return statement.

What is the reason for this log? Could it be more informative (in the logged message or tag)?